src/ngx_http_lua_headers.c

```diff
@@ -167,7 +167,7 @@ ngx_http_lua_ngx_req_raw_header(lua_State *L)
     size = 0;
     b = c->buffer;
 
-    if (mr->request_line.data[mr->request_line.len] == CR) {
+    if (mr->request_line.len>0 && mr->request_line.data[mr->request_line.len] == CR) {
```

Comment:
```suggestion
    if (mr->request_line.len == 0) {
        return 0;
    }

    if (mr->request_line.data[mr->request_line.len] == CR) {
```

Reply:
I'll run some tests with this proposal.

Reply:
Running this code produces the following error:

```
2022/01/31 15:53:18 [error] 9#0: *1 failed to run log_by_lua*: log_by_lua(nginx.conf:62):2: bad argument #1 to 'len' (string expected, got nil)
stack traceback:
        [builtin:len]: in function 'len'
        log_by_lua(nginx.conf:62):2: in main chunk while logging request, client: 172.17.0.1, server: localhost
```

I am unable to push new commits to this PR without a lengthy legal, security, and manager approval. I had to ask for an emergency escalation to get this PR logged in just four days. I expect future commits to take 1-2 weeks for a complete review.

I am happy to assist in testing any further commit proposals you may have.